### DISCUSSION TITLE
why GENERAL_NAME_free is undefined in libcrypto.a ?

### DISCUSSION THREAD

**Discussion by bug-code** [openssl/openssl#23071] [Q&A] [answered]
I try to build opensll 1.1.0 or 1.0.2 in ubuntu 22 , but i GENERAL_NAME_free is undefined in libcrypto.a
```bash
# nm -rCu libcrypto.a | grep GENERAL_NAME_free
        U GENERAL_NAME_free
```

**Comment by levitte** [comment, marked as answer]
It's not undefined in libcrypto.a as a whole, but is undefined in all object files that libcrypto.a includes, apart from the object file that defines it (libcrypto-lib-v3_genn.o, which is the compilation of crypto/x509/v3_genn.c).  This is normal, and is resolved by the linker when it links your software with libcrypto.a.

``` console
$ nm -gU libcrypto.a | grep GENERAL_NAME_free
00000000000001c7 T GENERAL_NAME_free
```

**Reply by bug-code** [reply]
thinks a lot ,  i figure it out
